protocol/contracts/amo/helpers/PoolHelper.sol

```diff
@@ -0,0 +1,248 @@
+pragma solidity ^0.8.4;
+// SPDX-License-Identifier: AGPL-3.0-or-later
+
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "../interfaces/AMO__IBalancerVault.sol";
+import "../helpers/AMOCommon.sol";
+
+
+interface IWeightPool2Tokens {
+    function getNormalizedWeights() external view returns (uint256[] memory);
+}
+
+contract PoolHelper is Ownable {
+    AMO__IBalancerVault public immutable balancerVault;
+    // @notice Temple price floor denominator
+    uint256 public constant TPF_PRECISION = 10_000;
+
+    // @notice temple index in balancer pool
+    uint64 public immutable templeBalancerPoolIndex;
+
+    // @notice percentage bounds (in bps) beyond which to rebalance up or down
+    uint64 public rebalancePercentageBoundLow;
+    uint64 public rebalancePercentageBoundUp;
+
+    bytes32 public immutable balancerPoolId;
+
+    // @notice Temple price floor ratio
+    TreasuryPriceFloor public templePriceFloorRatio;
+
+    struct TreasuryPriceFloor {
+        uint128 numerator;
+        uint128 denominator;
+    }
+
+    event SetTemplePriceFloorRatio(uint128, uint128);
+
+    constructor(
+      address _balancerVault,
+      address _temple,
+      bytes32 _balancerPoolId
+    ) {
+      (address[] memory tokens,,) = AMO__IBalancerVault(_balancerVault).getPoolTokens(_balancerPoolId);
+      uint256 index;
+      for (uint i=0; i<tokens.length; i++) {
+          if (tokens[i] == address(_temple)) {
+              index = i;
+              break;
+          }
+      }
+      templeBalancerPoolIndex = uint64(index);
+      balancerPoolId = _balancerPoolId;
+      balancerVault = AMO__IBalancerVault(_balancerVault);
+    }
+
+    function getBalances() public view returns (uint256[] memory balances) {
+      (, balances,) = balancerVault.getPoolTokens(balancerPoolId);
+    }
+
+    function setTemplePriceFloorRatio(uint128 _numerator) external onlyOwner {
+        templePriceFloorRatio.numerator = _numerator;
+        templePriceFloorRatio.denominator = uint128(TPF_PRECISION);
+
+        emit SetTemplePriceFloorRatio(_numerator, uint128(TPF_PRECISION));
+    }
+
+    // @notice percentage bounds (in bps) beyond which to rebalance up or down
+    function setRebalancePercentageBounds(uint64 belowTPF, uint64 aboveTPF) external onlyOwner {
+        if (belowTPF >= TPF_PRECISION || aboveTPF >= TPF_PRECISION) {
+            revert AMOCommon.InvalidBPSValue(belowTPF);
+        }
+        rebalancePercentageBoundLow = belowTPF;
+        rebalancePercentageBoundUp = aboveTPF;
+    }
+
+    function spotPriceUsingLPRatio() public view returns (uint256 templeBalance, uint256 stableBalance) {
+      uint256[] memory balances = getBalances();
+      if (templeBalancerPoolIndex == 0) {
```

Comment:
Can make this slightly less gas using ternary instead of if

```
(templeBalance, daiBalance) = (templeBalancerPoolIndex == 0) 
    ? (balances[0], balances[1]) 
    : (balances[1], balances[0]);
```